src/main/kotlin/com/openlattice/codex/controllers/CodexController.kt

```diff
@@ -38,98 +31,47 @@ import javax.inject.Inject
-                        .setStatusCallback(URI.create("https://api.openlattice.com/datastore/kodex/status")).create()
-                pendingTexts[message.sid] = message
-                processQueueEntry(message, organizationId)
-            }
-        }
-    }
+    private val twilioQueue = HazelcastQueue.TWILIO.getQueue(hazelcastInstance)
 
     @Timed
     @RequestMapping(path = ["", "/"], method = [RequestMethod.POST])
     override fun sendOutgoingText(@RequestBody contents: MessageRequest) {
+        ensureWriteAccess(AclKey(contents.messageEntitySetId))
+        contents.senderId = Principals.getCurrentUser().id
         twilioQueue.put(contents)
     }
 
-    override fun receiveIncomingText(@RequestBody message: Message) {
-
+    @Timed
+    @RequestMapping(path = [INCOMING + ORG_ID_PATH], method = [RequestMethod.POST], consumes = [MediaType.APPLICATION_FORM_URLENCODED_VALUE])
+    fun receiveIncomingText(@PathVariable(ORG_ID) organizationId: UUID, request: HttpServletRequest) {
+        codexService.processIncomingMessage(organizationId, request)
     }
 
     @Timed
-    @RequestMapping(path = [CodexApi.STATUS], method = [RequestMethod.POST])
-    override fun listenForTextStatus(@RequestBody message: Message) {
-        if (message.status == Message.Status.FAILED || message.status == Message.Status.UNDELIVERED) {
-            println("Message not received or even failed to send!!! ")
-        } else {
-            pendingTexts.remove(message.sid)
-        }
-    }
+    @RequestMapping(path = [INCOMING + ORG_ID_PATH + STATUS], method = [RequestMethod.POST])
+    fun listenForTextStatus(@PathVariable(ORG_ID) organizationId: UUID, request: HttpServletRequest) {
 
-    fun processQueueEntry(message: Message, organizationId: UUID) {
-        createEntitiesFromMessage(message, organizationId, peopleUUID)
-        createEntitiesFromMessage(message, organizationId, contactUUID)
-        createEntitiesFromMessage(message, organizationId, messagesUUID)
-        createEntitiesFromMessage(message, organizationId, toUUID)
-        createEntitiesFromMessage(message, organizationId, fromUUID)
+        val messageId = request.getParameter(CodexConstants.Request.SID.parameter)
+        val status = Message.Status.forValue(request.getParameter(CodexConstants.Request.STATUS.parameter))
 
-        createAssociationsFromMessage()
-    }
+        codexService.updateMessageStatus(organizationId, messageId, status)
 
-    fun createEntitiesFromMessage(msg: Message, organizationId: UUID, appTypeId: UUID) {
-        val entities = mutableListOf<Map<UUID, Set<Any>>>()
-        val app = appService.getApp("codex")
-        val ack = AppConfigKey(app.id, organizationId, appTypeId)
-        val esid = appConfigMS.load(ack).entitySetId
+        if (status == Message.Status.FAILED || status == Message.Status.UNDELIVERED) {
+            println("Message $messageId not received or even failed to send!!! ")
```

Comment:
Why not use a logger here?

Reply:
ah yeah -- that's old code that was just caught in the reformatting so I hadn't looked at it closely buuut idk why it's not using a logger